File: scripts/push-to-github.py
#!/usr/bin/env python3
import subprocess
import os

os.chdir('/vercel/share/v0-project')

try:
    # Check git status
    print("[v0] Checking git status...")
    result = subprocess.run(['git', 'status'], capture_output=True, text=True)
    print(result.stdout)
    
    # Add all changes
    print("[v0] Adding all changes...")
    subprocess.run(['git', 'add', '.'], check=True)
    
    # Commit changes
    print("[v0] Committing changes...")
    commit_msg = "fix: resolve deployment issues with TypeScript metadata and build configuration

- Remove ignoreBuildErrors setting to enable proper error reporting
- Add Metadata type imports to settings, credits, and tools pages
- Create middleware for proper authentication routing
- Enable Vercel to properly build and deploy updates"
    
    subprocess.run(['git', 'commit', '-m', commit_msg], check=True)
    
    # Push to the current branch
    print("[v0] Pushing to GitHub...")
    branch = subprocess.run(['git', 'rev-parse', '--abbrev-ref', 'HEAD'], capture_output=True, text=True, check=True)
    current_branch = branch.stdout.strip()
    print(f"[v0] Current branch: {current_branch}")
    
    result = subprocess.run(['git', 'push', 'origin', current_branch], capture_output=True, text=True)
    print(result.stdout)
    if result.stderr:
        print(result.stderr)
    
    print("[v0] ✓ Changes pushed successfully!")
    
except subprocess.CalledProcessError as e:
    print(f"[v0] Error: {e}")
    print(e.stdout)
    print(e.stderr)
except Exception as e:
    print(f"[v0] Error: {e}")

File: scripts/push-to-github.mjs
import { execSync } from 'child_process';
import { resolve } from 'path';

const projectDir = '/vercel/share/v0-project';

try {
  console.log('[v0] Starting deployment push...');
  
  // Set git user for commits
  execSync('git config user.email "v0[bot]@users.noreply.github.com"', { cwd: projectDir });
  execSync('git config user.name "v0[bot]"', { cwd: projectDir });
  
  // Check git status
  const status = execSync('git status --short', { cwd: projectDir }).toString();
  console.log('[v0] Git status:\n', status);
  
  // Add all changes
  console.log('[v0] Adding changes...');
  execSync('git add .', { cwd: projectDir });
  
  // Commit changes
  console.log('[v0] Committing changes...');
  const commitMessage = 'fix: resolve Vercel deployment issues\n\nRemove ignoreBuildErrors and fix TypeScript metadata exports\n- Removed ignoreBuildErrors: true from next.config.mjs to enable proper error reporting\n- Added Metadata type imports to settings, credits, and tools pages\n- Created middleware proxy for authentication routing\n\nThis enables Vercel to properly report and deploy updates.';
  
  execSync(`git commit -m "${commitMessage.split('\n')[0]}" --allow-empty`, { cwd: projectDir });
  
  // Get current branch
  const branch = execSync('git rev-parse --abbrev-ref HEAD', { cwd: projectDir }).toString().trim();
  console.log('[v0] Current branch:', branch);
  
  // Push to origin
  console.log('[v0] Pushing to GitHub...');
  execSync(`git push origin ${branch}`, { cwd: projectDir });
  
  console.log('[v0] ✅ Successfully pushed changes to GitHub!');
  console.log('[v0] Vercel will now detect the changes and deploy the updates automatically.');
  
} catch (error) {
  console.error('[v0] Error:', error.message);
  process.exit(1);
}
